app/ide-desktop/client/tests/electronTest.ts

```diff
@@ -1,57 +1,95 @@
 /** @file Commonly used functions for electron tests */
+/* eslint-disable no-empty-pattern */
 
-import { _electron, ElectronApplication, expect, type Page, test } from '@playwright/test'
+import { test as base, expect } from '@chromatic-com/playwright'
+import { _electron, ElectronApplication, type Page } from '@playwright/test'
 import { TEXTS } from 'enso-common/src/text'
-import * as random from 'lib0/random'
+import fs from 'node:fs/promises'
 import os from 'node:os'
-import pathModule from 'node:path'
+import path from 'node:path'
 
 const LOADING_TIMEOUT = 10000
 const TEXT = TEXTS.english
 export const CONTROL_KEY = os.platform() === 'darwin' ? 'Meta' : 'Control'
 
+const electronExecutablePath = await (async () => {
+  const POSSIBLE_EXEC_PATHS = [
+    '../../../../dist/ide/linux-unpacked/enso',
+    '../../../../dist/ide/win-unpacked/Enso.exe',
+    '../../../../dist/ide/mac/Enso.app/Contents/MacOS/Enso',
+    '../../../../dist/ide/mac-arm64/Enso.app/Contents/MacOS/Enso',
+  ].map((p) => path.resolve(import.meta.dirname, p))
+  try {
+    const promises = POSSIBLE_EXEC_PATHS.map((p) => fs.access(p, fs.constants.X_OK).then(() => p))
+    return await Promise.any(promises)
+  } catch {
+    throw Error('Cannot find Enso package')
+  }
+})();
+
 /**
  * Tests run on electron executable.
  *
  * Similar to playwright's test, but launches electron, and passes Page of the main window.
  */
-export function electronTest(
-  name: string,
-  body: (args: {
-    page: Page
-    app: ElectronApplication
-    projectsDir: string
-  }) => Promise<void> | void,
-) {
-  test(name, async () => {
-    const uuid = random.uuidv4()
-    const projectsDir = pathModule.join(os.tmpdir(), 'enso-test-projects', `${name}-${uuid}`)
-    console.log('Running Application; projects dir is', projectsDir)
+export const test = base.extend<{
+  testRunId: string,
+  projectsDir: string
+  app: ElectronApplication
+  page: Page
+}>({
+  testRunId: async function({}, use, testInfo) {
+    await use(`${testInfo.testId}-${new Date().toISOString()}`)
+  },
+  projectsDir: async function ({ testRunId }, use) {
+    const projectsDir = path.join(os.tmpdir(), 'enso-test-projects', testRunId)
+    await use(projectsDir)
+  },
+
+  /**
+   * Setup for all tests: checks if and where electron exec is.
+   * @throws when no Enso package could be found.
+   */
```

Comment:
Is this docs still relevant? To me all fields (`testRunId`, `projectsDir`) have something from being "setup"